db/migrate/20160318213755_create_gem_download.rb

```diff
@@ -0,0 +1,10 @@
+class CreateGemDownload < ActiveRecord::Migration
+  def change
+    create_table :gem_downloads do |t|
+      t.integer :rubygem_id, null: false
+      t.integer :version_id, null: false
```

Comment:
@arthurnn This schema feels very specific to the current count implementations. And using `0` values for gem & global totals isn't obvious. And the `rubygem_id` association is redundant when there's a `version_id`.

What do you think about replacing `rubygem_id` and `version_id` with a polymorphic `downloadable` assoc?

If the `downloadable` is a rubygem, is the total count for that gem. If it's a version, it's that version's count. And if it's nil, or a magic value, it's the global download count.